.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,315 @@
+      matrix:
+        build: [
+          {runs-on: self-hosted, enable_perf: OFF, name: "run"},
+          {runs-on: self-hosted, enable_perf: ON, name: "perf"},
+        ]
+
+    runs-on: ${{ matrix.build.runs-on }}
+
+    container:
+      image: ghcr.io/${{ github.repository }}/tt-mlir-ubuntu-22-04:latest
+      options: --user root
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+          fetch-depth: 0
+
+    - name: Set reusable strings
+      id: strings
+      shell: bash
+      run: |
+        echo "work-dir=$(pwd)" >> "$GITHUB_OUTPUT"
+        echo "build-output-dir=$(pwd)/build" >> "$GITHUB_OUTPUT"
+
+    - name: Git safe dir
+      run: git config --global --add safe.directory ${{ steps.strings.outputs.work-dir }}
+
+    - name: Use build artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: build-artifacts-${{ matrix.build.name }}
+        path: build
+
+    - name: Config compiler
+      shell: bash
+      run: |
+        source env/activate
+        cmake -G Ninja \
+        -B build \
+        -DCMAKE_CXX_COMPILER=clang++-17 \
+        -DCMAKE_C_COMPILER=clang-17 \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DTTMLIR_ENABLE_RUNTIME=ON \
+        -DTTMLIR_ENABLE_RUNTIME_TESTS=ON \
+        -DTT_RUNTIME_ENABLE_PERF_TRACE=${{ matrix.build.enable_perf }} \
+        -DTTMLIR_ENABLE_STABLEHLO=ON \
+
+    - name: (Host) Run tests
+      shell: bash
+      run: |
+        source env/activate
+        cmake --build build -- check-ttmlir
+
+    - name: Download n150 system descriptor
+      uses: actions/download-artifact@v4
+      with:
+        name: system-desc-n150
+
+    - name: (n150) Run tests
+      shell: bash
+      run: |
+        source env/activate
+        export SYSTEM_DESC_PATH="${GITHUB_WORKSPACE}/system_desc.ttsys"
+        cmake --build build -- check-ttmlir
+
+    - name: (n150) Upload test artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: n150-Silicon-${{ matrix.build.name }}
+        path: build/test/ttmlir/Silicon/
+
+    - name: Download n300 system descriptor
+      uses: actions/download-artifact@v4
+      with:
+        name: system-desc-n300
+
+    - name: (n300) Run tests
+      shell: bash
+      run: |
+        source env/activate
+        export SYSTEM_DESC_PATH="${GITHUB_WORKSPACE}/system_desc.ttsys"
+        cmake --build build -- check-ttmlir
+
+    - name: (n300) Upload test artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: n300-Silicon-${{ matrix.build.name }}
+        path: build/test/ttmlir/Silicon/
+
+  run-tests:
+    needs: generate-tests
+    strategy:
+      fail-fast: false
+      matrix:
+        build: [
+          {runs-on: n150, name: "run"},
+          #{runs-on: n150, name: "perf"}, #issue-512
+          {runs-on: n300, name: "run"},
+          #{runs-on: n300, name: "perf"}, #issue-512
+        ]
+
+    runs-on:
+      - self-hosted
+      - ${{ matrix.build.runs-on }}
```

Comment:
I think I'm missing how the `runs-on` n150 is the same runner that the silicon desc was harvested on.  It seems like we could get a system desc from runner `n150-3` and then try to run it on `n150-1`.

Reply:
you're right. I had incorrectly assumed it would land on the same machine since its defined in the same workflow file.

Reply:
An alternative way is to make the 2-phase CI
1) on compute runner build project and ttrt, save as artifacts
2) on silicon runner generate descriptors, compile test and run them

Reply:
yeah the more I think about this, the more I feel like we should make runners "stateless" ie we shouldn't have to depend on a specific runner downstream. I think it might be fine to just build all the tests and test on the runner itself instead of trying to upload it to artifacts. We can have the "runs-on" either specify a target family (ie n150) in which it will run on any machine that has the tag n150, or have other runs-on (target silicon) in which it has to run on a specific harvested test machine

Reply:
fixed workflow file. It will act in two stages when executing